gulp-core-build-karma/src/KarmaTask.ts

```diff
@@ -67,6 +78,30 @@ export class KarmaTask extends GulpTask<IKarmaTaskConfig> {
 
       completeCallback();
     } else {
+      // Normalize the match expression if one was specified
+      const { testMatch } = this.taskConfig; // tslint:disable-line:typedef
+      if (testMatch) {
+        let normalizedMatch: RegExp;
+        if (typeof testMatch === 'string') {
+          normalizedMatch = new RegExp(testMatch as string);
+        } else if (testMatch instanceof RegExp) {
+          normalizedMatch = testMatch;
+        } else {
+          completeCallback('The testMatch regular expression is invalid');
+          return;
+        }
+
+        // tslint:disable:max-line-length
+        const testsJsFileContents: string = [
+          `var context = require.context('${path.posix.join('..', this.buildConfig.libFolder)}', true, ${normalizedMatch.toString()});`,
+          `context.keys().forEach(context);`,
+          `module.exports = context;`
+        ].join(os.EOL);
+        // tslint:enable:max-line-length
+
+        fs.writeFileSync(path.join(this.buildConfig.tempFolder, 'tests.js'), testsJsFileContents);
```

Comment:
Ew. To write one file? Massive overkill to use gulp stream for that, no? I almost never use streams unless reading lots of files.